lib/dashboard/DefaultDashboardFactory.ts

```diff
@@ -108,52 +122,80 @@ export class DefaultDashboardFactory
       );
     }
 
-    const renderingPreference =
-      props.renderingPreference ??
-      DashboardRenderingPreference.INTERACTIVE_ONLY;
-    const detailStart: string =
-      "-" + (props.detailDashboardRange ?? Duration.hours(8)).toIsoString();
-    const summaryStart: string =
-      "-" + (props.summaryDashboardRange ?? Duration.days(14)).toIsoString();
-    let anyDashboardCreated = false;
-
-    if (createDashboard) {
-      anyDashboardCreated = true;
-      this.dashboard = this.createDashboard(renderingPreference, "Dashboard", {
-        dashboardName: props.dashboardNamePrefix,
-        start: detailStart,
-        periodOverride:
-          props.detailDashboardPeriodOverride ?? PeriodOverride.INHERIT,
+    if (props.dashboardTypes) {
+      // Dashboard types are defined, so use new dynamic dashboarding features
+      const renderingPreference =
+        props.renderingPreference ??
+        DashboardRenderingPreference.INTERACTIVE_ONLY;
+      const start: string = "-" + Duration.hours(8).toIsoString();
+
+      console.log("Creating dashboards for types = " + props.dashboardTypes);
+      props.dashboardTypes?.forEach((dashboardType) => {
+        const dashboard = this.createDashboard(
+          renderingPreference,
+          dashboardType,
+          {
+            dashboardName: `${props.dashboardNamePrefix}-${dashboardType}`,
+            start: start,
+            periodOverride:
+              props.detailDashboardPeriodOverride ?? PeriodOverride.INHERIT,
+          }
+        );
+
+        this.dashboards[dashboardType] = dashboard;
```

Comment:
You could do this within `this.createDashboard`, so the old dashboards are effectively just presets and you can also get them the same way.

Reply:
This should be moot now that I've moved the new functionality to a new `DefaultDynamicDashboardFactory` class.